File: Jan-2023/C++/MinimumNumberOfArrowsToBurstBalloons.cxx
// 452. Minimum Number of Arrows to Burst Balloons 

class Solution {
public:
    bool static compareFunction(vector<int> &a, vector<int> &b) {
        return a[1]<b[1];
      }
    int findMinArrowShots(vector<vector<int>>& points) {
        
        //bool compareFunction(const vector<int> &a, const vector<int> &b){
           // return a[1]>b[1];
       // }
        sort(points.begin(), points.end(), compareFunction);
        
        int arrowCount = 1;
        int range = points[0][1];
        
        for(int i=1;i<points.size();i++){
            if(points[i][0]<=range){
                continue;
            }
            else{
                arrowCount++;
                range = points[i][1];
            }
             
        }
        return arrowCount;
    }
};

File: Jan-2023/Java/MinimumNumberOfArrowsToBurstBalloons.java
// 452. Minimum Number of Arrows to Burst Balloons 

class Solution {

    public int findMinArrowShots(int[][] points) {

        //Integer overflow fix

        //compares two elements, instead of subtracting which causes integer overflow

        Arrays.sort(points,(a,b)->Integer.compare(a[1],b[1]));

        //range, all balloons between this end variable will be popped

       int end = points[0][1];

       int arrowCount = 1;

       for(int i=1;i<points.length;i++){

          if(points[i][0] <= end){

               continue;

               //do nothing

           }

           else{

               //used one arrow

               arrowCount++;

               //update range

               end = points[i][1];

           }

       }

        return arrowCount;

    }

}
